tests/pyensign/test_connection.py

```diff
@@ -306,38 +311,49 @@ async def record_acks(ack):
 
     async def test_subscribe(self, client):
         topic_ids = [str(ULID()), str(ULID())]
-        events = 0
-        async for rep in client.subscribe(topic_ids):
-            assert isinstance(rep, event_pb2.Event)
-            events += 1
-            if events == 2:
-                break
-
-        # Should be able to resume subscribing to an existing stream.
-        async for rep in client.subscribe(topic_ids):
-            assert isinstance(rep, event_pb2.Event)
-            events += 1
-
+        event_ids = []
+        acked = asyncio.Event()
+
+        async def ack_event(event):
+            nonlocal event_ids
+            assert isinstance(event, Event)
+            await event.ack()
+            event_ids.append(event.id)
+            if len(event_ids) >= 2:
+                acked.set()
+
+        # Subscribe using an event callback.
+        await client.subscribe(topic_ids, ack_event)
+
+        # Wait for the callback to ack all the events.
+        await acked.wait()
         await client.close()
-        assert events == 3
+        assert len(event_ids) == 2
```

Comment:
What made this drop from three to two events?

Reply:
A typo, nice catch.